thunder/executors/pythonex.py

```diff
@@ -202,6 +202,13 @@ def _elementwise_unary_checker(x: NumberLike | TensorProxy) -> bool:
     return isinstance(x, (Number, NumberProxy))
 
 
+# Maps exact inputs to truncation division
+def _div_prim_impl(a: Number, b: Number) -> torch.Tensor:
+    if isinstance(a, (int, bool)) and isinstance(b, (int, bool)):
+        return a // b
```

Comment:
c-style integer division is truncation division, not floor division, so I think the implementation needs to be like this:

```
def _div_numbers(a: Number, b: Number) -> Number:
    if dtypes.is_exact_dtype(type(a)) and dtypes.is_exact_dtype(type(b)):
        # Accounts for rounding towards zero instead of flooring
        if (a >= 0) != (b >= 0) and a % b:
            return a // b + 1
        else:
            return a // b

    return a / b
```

Reply:
I updated the python executor implementation to match the prim number function.

Reply:
@mruberry, why do we need to follow C-style division? There's going to be a discrepancy in the results computed from the Thunder-jitted function and its Python original.

Reply:
The Python "division" operation is true division and we map it to the appropriate primitive here:

https://github.com/Lightning-AI/lightning-thunder/blob/63b01d58af1e7b11acb8e60ad1694e59814eb90e/thunder/core/proxies.py#L721

The `div` prim implements C-style division. We can rename it to `cdiv` if that would be clearer. If we do that, we probably shouldn't name a primitive `div`, just as Python no longer recommends setting `__div__` in favor of the explicit true division and floor division dunders.